### PR TITLE
Check transaction_id in TClearTransactionTaskRequest

### PR DESCRIPTION
**Pull request by chaoyli**

### CODE REVIEW

File: be/src/agent/task_worker_pool.cpp [hunk 1]
```diff
@@ -960,8 +960,10 @@ void* TaskWorkerPool::_clear_transaction_task_worker_thread_callback(void* arg_t
         vector<string> error_msgs;
         TStatus task_status;
 
-        worker_pool_this->_env->storage_engine()->clear_transaction_task(
-            clear_transaction_task_req.transaction_id, clear_transaction_task_req.partition_id);
+        if (clear_transaction_task_req.transaction_id > 0) {
```
Comment (imay) [inline be/src/agent/task_worker_pool.cpp]: Add comment to explain why there will be transaction id less than 0.

File: be/src/agent/task_worker_pool.cpp [hunk 2]
```diff
@@ -960,8 +960,10 @@ void* TaskWorkerPool::_clear_transaction_task_worker_thread_callback(void* arg_t
...
+            worker_pool_this->_env->storage_engine()->clear_transaction_task(
+                    clear_transaction_task_req.transaction_id, clear_transaction_task_req.partition_id);
+        }
         LOG(INFO) << "finish to clear transaction task. signature:" << agent_task_req.signature
```
Comment (imay) [inline be/src/agent/task_worker_pool.cpp]: Why don't log in if block?

**Review by imay — CHANGES_REQUESTED** [review]

**Comment by chaoyli** [comment]
#1876

File: be/src/agent/task_worker_pool.cpp (hunk 1 above)
Comment (chaoyli) [reply be/src/agent/task_worker_pool.cpp]: OK

File: be/src/agent/task_worker_pool.cpp (hunk 2 above)
Comment (chaoyli) [reply be/src/agent/task_worker_pool.cpp]: OK

**Review by imay — DISMISSED** [review]
LGTM

File: be/src/agent/task_worker_pool.cpp [hunk 3]
```diff
@@ -960,10 +960,15 @@ void* TaskWorkerPool::_clear_transaction_task_worker_thread_callback(void* arg_t
...
-                  << ", transaction_id:" << clear_transaction_task_req.transaction_id;
+        if (clear_transaction_task_req.transaction_id > 0) {
+            // transaction_id should be greater than zero.
+            // If it is not greater than zero, no need to execute
+            // the following clear_transaction_task() function.
+            worker_pool_this->_env->storage_engine()->clear_transaction_task(
+                    clear_transaction_task_req.transaction_id, clear_transaction_task_req.partition_id);
+            LOG(INFO) << "finish to clear transaction task. signature:" << agent_task_req.signature
+                      << ", transaction_id:" << clear_transaction_task_req.transaction_id;
+        }
```
Comment (morningman) [inline be/src/agent/task_worker_pool.cpp]: I think we should add an `else` block to print warning log if transaction id < 0.
This still a bug.

**Review by morningman — CHANGES_REQUESTED** [review]

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**